### PR TITLE
OR queries more validation

### PR DESCRIPTION
**Pull request by MarkDuckworth**
OR queries more validation

### CODE REVIEW

File: packages/firestore/src/lite-api/query.ts [hunk 1]
```diff
@@ -168,10 +168,12 @@ export function query<T>(
 
   if (nonFilters !== undefined) {
     queryConstraints = queryConstraints.concat(
-      nonFilters as QueryNonFilterConstraint[]
+      nonFilters as AppliableConstraint[]
```
Comment (ehsannas) [inline packages/firestore/src/lite-api/query.ts:171]: I realize it's not a change made in this PR, but I just noticed it:
it looks like there's a bug on line 166 above (the return value of concat is ignored)?

**Review by ehsannas — APPROVED** [review]

File: packages/firestore/src/lite-api/query.ts [hunk 2]
```diff
@@ -154,24 +155,22 @@ export function query<T>(
...
 export function query<T>(
   query: Query<T>,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  filterOrQueryConstraints: any,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  ...nonFilters: any
+  filterOrQueryConstraint:
+    | QueryCompositeFilterConstraint
+    | QueryConstraint
+    | undefined,
```
Comment (ehsannas) [inline packages/firestore/src/lite-api/query.ts:161]: why do we need `| undefined` here?

Comment (MarkDuckworth) [reply packages/firestore/src/lite-api/query.ts:161]: `filterOrQueryConstraint` can be undefined in this case `query(collection)`. The code would also work if the parameter is declared optional `filterOrQueryConstraint?`, but it also works if the parameter is not declared optional or `| undefined`. Out of the three, I wanted to be explicit, but could also go with making it optional...

Comment (ehsannas) [reply packages/firestore/src/lite-api/query.ts]: 👍

**Review by ehsannas — APPROVED** [review]

**Pull request merged**